lua/nvim-navic/init.lua

```diff
@@ -498,12 +498,49 @@ function M.get_location(opts)
 		end
 	end
 
-	if local_config.depth_limit ~= 0 and #location > local_config.depth_limit then
-		location = vim.list_slice(location, #location - local_config.depth_limit + 1, #location)
-		if local_config.highlight then
-			table.insert(location, 1, "%#NavicSeparator#" .. local_config.depth_limit_indicator .. "%*")
+	if type(local_config.depth_limit) == "function" then
+		local context = {
+			winid = vim.api.nvim_get_current_win(),
+			path = vim.fn.expand('%:p'),
+			mod = vim.bo.mod,
+		}
+		if type(local_config.depth_limit(context)) == "number" then
+			local start_index = 0
+			while
+				table.concat(
+					vim.list_slice(location, start_index + 1, #location),
+					(local_config.highlight and "%#NavicSeparator#" or '')
+					.. local_config.separator
+					.. (local_config.highlight and "%*" or ""))
+				:gsub('%%#%a+#', '')
+				:gsub('%%', ''):gsub('%*', '')
+				:len() > local_config.depth_limit(context)
```

Comment:
Can you briefly explain whats happening here

Reply:
The point is that it's hard for a user to create any reasonable while-loop since they don't have access to the `location` list. Let me give you an example:

```
[1, 2, 3, ..., 20] -> [5, 6, 7, ..., 20] // I can start from `20` since I can access `location`
```

But with the current `depth_limit` as a `number`, I have to do:

```
[1, 2, 3, ..., 50] // I have to guess a number that is large enough.
<----------------> // depth_limit = 50
[2, 3, ..., 50]
<-------------> // depth_limit = 49
...
[5, ..., 50]
<----------> // depth_limit = 46
```

and I have to assume that your original implementation can work with `depth_limit > #location`.

```lua
(local_config.highlight and "%#NavicSeparator#" or '')
					.. local_config.separator
					.. (local_config.highlight and "%*" or ""))
				:gsub('%%#%a+#', '')
				:gsub('%%', ''):gsub('%*', '')
```

All these lines basically nonsense after I realized that you have provided `get_location({ highlight = false })`. (but this doesn't resolve my example above.)